### PR TITLE
[SPARK-34630][PYTHON][FOLLOWUP] Add __version__ into pyspark init __all__

### PR DESCRIPTION
**Pull request by Yikun**
### What changes were proposed in this pull request?
This patch add `__version__` into pyspark.__init__.__all__ to make the `__version__` as exported explicitly, see more in https://github.com/apache/spark/pull/32110#issuecomment-817331896

### Why are the changes needed?
1. make the `__version__` as exported explicitly
2. cleanup `noqa: F401` on `__version`

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Python related CI passed

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @zero323

File: python/pyspark/__init__.py [hunk 1]
```diff
@@ -60,7 +60,7 @@
 from pyspark.serializers import MarshalSerializer, PickleSerializer
 from pyspark.taskcontext import TaskContext, BarrierTaskContext, BarrierTaskInfo
 from pyspark.profiler import Profiler, BasicProfiler
-from pyspark.version import __version__  # noqa: F401
+from pyspark.version import __version__ as __version__
```
Comment (zero323) [inline python/pyspark/__init__.py:63]: Technically speaking, we just need 

```python
from pyspark.version import __version__
```

here.

Following

```python
from pyspark.version import __version__ as __version__
```

should be placed in `__init__.pyi` to mark [explicit re-export](https://mypy.readthedocs.io/en/stable/command_line.html?highlight=export#cmdoption-mypy-no-implicit-reexport).

We should also drop https://github.com/apache/spark/blob/c3f4763282567a2af4a2ed88726af27e7fd3c439/python/pyspark/__init__.pyi#L75

Comment (Yikun) [reply python/pyspark/__init__.py]: Done

**Review by srowen — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/__init__.pyi [hunk 2]
```diff
@@ -72,4 +72,4 @@ def copy_func(
 ) -> F: ...
 def keyword_only(func: F) -> F: ...
 
-__version__: str
+from pyspark.version import __version__ as __version__
```
Comment (zero323) [inline python/pyspark/__init__.pyi:75]: Nit-pick: Could you move this to the top of the file, next to the other imports?

**Review by zero323 — APPROVED** [review]
LGTM, excluding position of the import in the stub.

File: python/pyspark/__init__.pyi (hunk 2 above)
Comment (Yikun) [reply python/pyspark/__init__.pyi]: oops, done. : )

**Review by zero323 — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
I will leave it to you @zero323 to merge :-).

**Comment by zero323** [comment]
Merged to master and branch-3.1.

Thanks everyone!

**Pull request closed without merging**